azurelinuxagent/metadata.py

```diff
@@ -70,24 +71,12 @@ def get_distro():
 Add this walk arround for detecting Snappy Ubuntu Core temporarily, until ubuntu 
 fixed this bug: https://bugs.launchpad.net/snappy/+bug/1481086
 """
-def which(program):
```

Comment:
there's a problem of this method.
On a normal Ubuntu 15.04 VM, user could install package snappy-tools to build snap package. If snappy-tools is installed, which("snappy") will return true, and it will consider the VM as a snappy VM